tests/migrations/test_operations.py

```diff
@@ -501,6 +501,56 @@ def test_create_model_with_constraint(self):
         self.assertEqual(definition[1], [])
         self.assertEqual(definition[2]['options']['constraints'], [check_constraint])
 
+    def test_create_model_with_partial_unique_constraint(self):
+        partial_unique_constraint = models.UniqueConstraint(
+            fields=["pink"],
+            condition=models.Q(weight__gt=5),
+            name="test_constraint_pony_pink_for_weight_gt_5_uniq",
+        )
+        operation = migrations.CreateModel(
+            "Pony",
+            [
+                ("id", models.AutoField(primary_key=True)),
+                ("pink", models.IntegerField(default=3)),
+                ("weight", models.FloatField()),
+            ],
+            options={"constraints": [partial_unique_constraint]},
+        )
+
+        # Test the state alteration
+        project_state = ProjectState()
+        new_state = project_state.clone()
+        operation.state_forwards("test_crmo", new_state)
+        self.assertEqual(len(new_state.models["test_crmo", "pony"].options["constraints"]), 1)
+
+        # Test database alteration
+        self.assertTableNotExists("test_crmo_pony")
+        with connection.schema_editor() as editor:
+            operation.database_forwards("test_crmo", editor, project_state, new_state)
+        self.assertTableExists("test_crmo_pony")
+
+        # Test constraint actually works
+        with connection.cursor() as cursor:
+            cursor.execute("INSERT INTO test_crmo_pony (id, pink, weight) VALUES (1, 1, 4.0)")
```

Comment:
Did you try the ORM rather than raw SQL? I'm not sure if there's a reason the test above doesn't use it.

Reply:
changed to ORM